pkg/virt-handler/vm.go

```diff
@@ -291,6 +292,12 @@ func (d *VMHandlerDispatch) processVmUpdate(vm *v1.VM, shouldDeleteVm bool) erro
 		return err
 	}
 
+	// Map Container Registry Disks to block devices Libvirt can consume
+	vm, err = registrydisk.MapRegistryDisks(vm)
+	if err != nil {
+		return err
+	}
```

Comment:
Right, so I don't exactly disagree or agree with your thoughts on the REST endpoint quite yet.  However, I think the REST endpoint is out of the scope of what I'm trying to do here though. 

This comes to the point I made at the top of this PR about this feature's lack of interference with your storage proposal #311.

I implemented this feature in a way that is most natural for KubeVirt as it is today. If you look about 3 or 4 lines up you'll see we map PVCs to specific technology backends. These are established patterns I'm using with clear entry points that are isolated to the ContainerRegistryDisk feature.

What I don't want to see is for this feature to be blocked by the much larger storage discussion.   

If we decide to move to a generic REST api for retrieving storage devices, then the ContainerRegistryDisk feature will migrate to use that logic.  I purposefully encapsulated the logic in this feature to make this type of transition simple for us.